### ISSUE TITLE
Unable to create more admin users after initial user has been created

### ISSUE THREAD

**Issue by devillex** [orionjs/orion#239]
After setting up Orion I navigate to .../admin and register my first user. This user is automatically made an admin user, as expected.

When trying to create another admin user I get an "Internal Server Error" dialog, and the following error in the console:
```
I20150720-21:47:17.433(2)? Exception while invoking method 'accountsCreateUser' MongoError: insertDocument :: caused by :: 11000 E11000 duplicate key error index: meteor.roles.$name_1  dup key: { : null }
I20150720-21:47:17.433(2)?     at Object.Future.wait (/home/devillex/.meteor/packages/meteor-tool/.1.1.3.4sddkj++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:398:15)
I20150720-21:47:17.433(2)?     at [object Object]._.extend.wrapAsync (packages/meteor/helpers.js:119:1)
I20150720-21:47:17.433(2)?     at [object Object]._.each.MongoConnection.(anonymous function) [as update] (packages/mongo/mongo_driver.js:678:1)
I20150720-21:47:17.433(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) (packages/mongo/collection.js:575:1)
I20150720-21:47:17.434(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) [as update] (packages/aldeed:collection2/collection2.js:165:1)
I20150720-21:47:17.434(2)?     at [object Object].Mongo.Collection.upsert (packages/mongo/collection.js:609:1)
I20150720-21:47:17.434(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) [as upsert] (packages/aldeed:collection2/collection2.js:165:1)
I20150720-21:47:17.434(2)?     at Object.Roles.addUserToRoles (packages/nicolaslopezj:roles/roles_server.js:23:1)
I20150720-21:47:17.434(2)?     at Object.orion.adminExists.Roles._collection.find.roles (packages/orionjs:accounts/accounts_server.js:12:1)
I20150720-21:47:17.434(2)?     at CollectionHooks.defineAdvice.args.(anonymous function) (packages/matb33:collection-hooks/insert.js:30:1)
I20150720-21:47:17.434(2)?     - - - - -
I20150720-21:47:17.434(2)?     at Object.toError (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/utils.js:114:11)
I20150720-21:47:17.434(2)?     at updateWithWriteCommands (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/collection/core.js:577:27)
I20150720-21:47:17.434(2)?     at Db.command (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/db.js:1195:7)
I20150720-21:47:17.434(2)?     at Db._executeQueryCommand (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/db.js:1903:9)
I20150720-21:47:17.435(2)?     at Server.Base._callHandler (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/base.js:453:41)
I20150720-21:47:17.435(2)?     at Server.connect.connectionPool.on.server._serverState (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/server.js:487:18)
I20150720-21:47:17.435(2)?     at [object Object].MongoReply.parseBody (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
I20150720-21:47:17.435(2)?     at [object Object].Server.connect.connectionPool.on.server._serverState (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/server.js:445:20)
I20150720-21:47:17.435(2)?     at [object Object].emit (events.js:95:17)
I20150720-21:47:17.435(2)?     at [object Object]._connect (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:207:13)
```

When I navigate to the Accounts page I see that the account was in fact created, but the 'Name' of the user was not set, and they are also not assigned to the 'admin' role.

I try to assign the 'admin' role to the user on via their account page, but again I get an 'Internal Server Error' dialog, but a different exception on the console:
```
I20150720-21:54:13.410(2)? Exception while invoking method 'updateRoles' MongoError: insertDocument :: caused by :: 11000 E11000 duplicate key error index: meteor.roles.$name_1  dup key: { : null }
I20150720-21:54:13.410(2)?     at Object.Future.wait (/home/devillex/.meteor/packages/meteor-tool/.1.1.3.4sddkj++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:398:15)
I20150720-21:54:13.410(2)?     at [object Object]._.extend.wrapAsync (packages/meteor/helpers.js:119:1)
I20150720-21:54:13.410(2)?     at [object Object]._.each.MongoConnection.(anonymous function) [as update] (packages/mongo/mongo_driver.js:678:1)
I20150720-21:54:13.410(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) (packages/mongo/collection.js:575:1)
I20150720-21:54:13.411(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) [as update] (packages/aldeed:collection2/collection2.js:165:1)
I20150720-21:54:13.411(2)?     at [object Object].Mongo.Collection.upsert (packages/mongo/collection.js:609:1)
I20150720-21:54:13.411(2)?     at [object Object]._.each.Mongo.Collection.(anonymous function) [as upsert] (packages/aldeed:collection2/collection2.js:165:1)
I20150720-21:54:13.411(2)?     at Object.Roles.setUserRoles (packages/nicolaslopezj:roles/roles_server.js:35:1)
I20150720-21:54:13.411(2)?     at [object Object].Meteor.methods.updateRoles (packages/orionjs:accounts/accounts-tab/server.js:43:1)
I20150720-21:54:13.411(2)?     at maybeAuditArgumentChecks (packages/ddp/livedata_server.js:1617:1)
I20150720-21:54:13.411(2)?     - - - - -
I20150720-21:54:13.411(2)?     at Object.toError (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/utils.js:114:11)
I20150720-21:54:13.411(2)?     at updateWithWriteCommands (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/collection/core.js:577:27)
I20150720-21:54:13.411(2)?     at Db.command (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/db.js:1195:7)
I20150720-21:54:13.411(2)?     at Db._executeQueryCommand (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/db.js:1903:9)
I20150720-21:54:13.411(2)?     at Server.Base._callHandler (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/base.js:453:41)
I20150720-21:54:13.412(2)?     at Server.connect.connectionPool.on.server._serverState (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/server.js:487:18)
I20150720-21:54:13.412(2)?     at [object Object].MongoReply.parseBody (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
I20150720-21:54:13.412(2)?     at [object Object].Server.connect.connectionPool.on.server._serverState (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/server.js:445:20)
I20150720-21:54:13.412(2)?     at [object Object].emit (events.js:95:17)
I20150720-21:54:13.412(2)?     at [object Object]._connect (/home/devillex/.meteor/packages/mongo/.1.1.0.zb7oxo++os+web.browser+web.cordova/npm/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:207:13)
```

I could have sworn that this worked before. The only think that I can think of that has changed is that I removed the 'insecure' package from the Meteor package list.

**Comment by rwatts3** [comment]
I would start by cloning the example repo. There you can assign the role of admin to any user you'd like upon logging into the panel.

**Comment by devillex** [comment]
Hi @rwatts3. I'm not sure how this will help me diagnose the problem and fix it on my existing project. Can you please elaborate a little on your solution?

**Comment by rwatts3** [comment]
Ok first things first let's cover a few bases.

1. Are you using the example repo , or did you start a project and add orion to it ? 
2. Have you gone through the tutorials, and looked at the documentation? 
3. Can you provide a link to a repo or branch where your project is located so that I may take a look at your code.

**Comment by devillex** [comment]
Thanks for the reply and your help!

Here are the answers to your questions:
1. I am busy with a Meteor project other than the example repo, and I added Orion to it afterwards by following the documentation.
2. I have worked through the documentation and looked at some tutorials.
3. Not sure how feasible this is, as the repo I'm working on is private. I'm sure I can provide specific parts of the code here if you indicate what would be beneficial.

**Comment by rwatts3** [comment]
Understood, 
Yes More specifically I would like to see how you are adding the second user to admin. 
If you go to the "Account" or "My Account" tab you will see the ability to add a user. When you add the user, and click on rolls, you should successfully have a secondary admin user. 

If this is not working you can create another role called lower-admin and extend the admin roll. 
To be honest you really only want 1 super admin user at a time. If you would like to give a user as many permissions as the admin user, you can custom define your roll and set permissions accordingly

**Comment by rwatts3** [comment]
I'll have to take a look at the code but i've never added a secondary "admin" user with all the same permissions as myself, I usually will create a lower admin or admin level two roll and give that user elevated permissions, but never the full permissions to the application.

**Comment by devillex** [comment]
Ah, what you are saying makes sense. I should create a second role with the appropriate permissions and then assign the other users to this role rather.

I guess I was a little confused by the error messages in the stack traces mentioned above. If it is a limitation / recommendation to have 1 admin user, then it may be a good idea to update the documentation and error messages to reflect this.

**Comment by devillex** [comment]
I finally found the problem.

I had this error:
```
Exception while invoking method 'updateRoles' MongoError: insertDocument :: caused by :: 11000 E11000 duplicate key error index: meteor.roles.$name_1  dup key: { : null }
```

Which was caused by this the following index from the [alanningmeteor-roles](https://github.com/alanning/meteor-roles) package, specifically the  [roles_server.js](https://github.com/alanning/meteor-roles/blob/master/roles/roles_server.js) file:
```
...
Meteor.roles._ensureIndex('name', {unique: 1})
...
```
I had the alanning:meteor package previously installed, but have since removed it. The index it created still remained.

I fixed the error removing this index via the MongoDB console, and issueing this command:
```
db.roles.dropIndex("name_1")
```

After this, I could once again assign other users to the 'admin' (or any other defined) role.
